docs/content/develop/resource.md

```diff
@@ -0,0 +1,455 @@
+   ```
+   cd ~/magic-modules
+   git checkout main && git clean -f . && git checkout -- . && git pull
+   cd $GOPATH/src/github.com/hashicorp/terraform-provider-google
+   git checkout main && git clean -f . && git checkout -- . && git pull
+   cd $GOPATH/src/github.com/hashicorp/terraform-provider-google-beta
+   git checkout main && git clean -f . && git checkout -- . && git pull
+   ```
+
+## Add a resource
+
+{{< tabs "resource" >}}
+{{< tab "MMv1" >}}
+1. Using an editor of your choice, in the appropriate [product folder]({{<ref "/get-started/how-magic-modules-works.md#mmv1" >}}), create a file called `RESOURCE_NAME.yaml`. Replace `RESOURCE_NAME` with the name of the API resource you are adding support for. For example, a configuration file for [NatAddress](https://cloud.google.com/apigee/docs/reference/apis/apigee/rest/v1/organizations.instances.natAddresses) would be called `NatAddress.yaml`.
+2. Copy the following template into the new file:
+   ```yaml
+   # Copyright 2023 Google Inc.
+   # Licensed under the Apache License, Version 2.0 (the "License");
+   # you may not use this file except in compliance with the License.
+   # You may obtain a copy of the License at
+   #
+   #     http://www.apache.org/licenses/LICENSE-2.0
+   #
+   # Unless required by applicable law or agreed to in writing, software
+   # distributed under the License is distributed on an "AS IS" BASIS,
+   # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   # See the License for the specific language governing permissions and
+   # limitations under the License.
+
+   --- !ruby/object:Api::Resource
+   # API resource name
+   name: 'ResourceName'
+   # Resource description for the provider documentation.
+   description: |
+     RESOURCE_DESCRIPTION
+   references: !ruby/object:Api::Resource::ReferenceLinks
+     guides:
+      # Link to quickstart in the API's Guides section. For example:
+      # 'Create and connect to a database': 'https://cloud.google.com/alloydb/docs/quickstart/create-and-connect'
+       'QUICKSTART_TITLE': 'QUICKSTART_URL'
+     # Link to the REST API reference for the resource. For example,
+     # https://cloud.google.com/alloydb/docs/reference/rest/v1/projects.locations.backups
+     api: 'API_REFERENCE_URL'
+   # Marks the resource as beta-only. Ensure a beta version block is present in
+   # provider.yaml.
+   # min_version: beta
+
+   # Inserts styled markdown into the header of the resource's page in the
+   # provider documentation.
+   # docs:
+   #   warning: WARNING_MARKDOWN
+   #   note: NOTE_MARKDOWN
+
+   # URL for the resource within the API domain. This should match the
+   # resource's create URL (excluding any query parameters).
+   # Terraform field names enclosed in double curly braces will be replaced
+   # with the field values from the resource.
+   base_url: 'projects/{{project}}/locations/{{location}}/resourcenames'
+   # URL for a created resource within the API domain. This should match
```

Comment:
Consider referencing AIPs. For example, "the URL for getting a single resource." -> "the URL of the standard GET method of the resource (https://google.aip.dev/131)"

Reply:
Updated these in b4424c449